lib/src/connector/connection.cc

```diff
@@ -338,7 +338,11 @@ std::string const& Connection::getWsUri() {
 }
 
 void Connection::switchWsUri() {
+    auto old_t = connection_target_.load();
     ++connection_target_;
+    auto current_t = connection_target_.load();
```

Comment:
This isn't a sufficient test, given that you might have only a single item and I used modulo to select from a list. See `getWsUri` right above this. You may want to instead compare URIs.